src/components/common/Main/cmp.tsx

```diff
@@ -1,17 +1,23 @@
 import { useEffect } from 'react'
 import { MainProps } from './types'
 import { useConnect } from '@/hooks/common/useConnect'
+import { useSessionStorage } from 'usehooks-ts'
 
 export const Main = ({ children }: MainProps) => {
   const { isConnected, connect } = useConnect()
+  const [keepAccountAlive] = useSessionStorage(
```

Comment:
I would move all this logic inside the useConnect hook, just exporting `tryRecconect` method that will handle this internally,

Reply:
Yeah having this logic dedupped accross two files did not make a lot of sense to me either. 
Changing that.